types/cookieclicker/index.d.ts

```diff
@@ -2533,6 +2559,93 @@ declare namespace Game {
 
     export function magicCpS(what: unknown): number;
     export let SpecialGrandmaUnlock: number;
+
+    type YouCustomizerAddonGeneId = 'face' | 'head' | 'hair' | 'acc1' | 'acc2';
+    type YouCustomizerColorGeneId = 'skinCol' | 'hairCol';
+    type YouCustomizerGeneId = YouCustomizerAddonGeneId | YouCustomizerColorGeneId;
+
+    class YouCustomizerAddonGene {
+        id: YouCustomizerAddonGeneId;
+        isList: true;
+        /**
+         * Default value.
+         */
+        def: number;
+        /**
+         * List of possible choices.
+         * Each choice is an offset in 'youAddons.png', similar to an icon.
+         */
+        choices: Array<[number, number]>;
+        /**
+         * Index of the gene in Game.YouCustomizer.genes.
+         */
+        n: number;
+    }
+    class YouCustomizerColorGene {
+        id: YouCustomizerAddonGeneId;
+        isList: true;
+        def: number;
+        choices: number[];
+    }
+
+    export let YouCustomizer: {
```

Comment:
You mean something like this?
```typescript
export interface YouCustomizerT: {
    ...
};
export let YouCustomizer: YouCustomizerT;
```
There will ever be only one `YouCustomizerT`, so I thought it made more sense to not explicitly define a separate interface for it.

Reply:
Yes, that's what I was thinking of, and while I do see the point of an external type being kinda pointless, to me, defining big types like that inline seems like a bad pattern. But also you never know, maybe some mod might have a second customizer kind of thing! In any case, this isn't a major problem, since the type can be (with your initial typings) extracted as `typeof Game["YouCustomizer"]`, so I suppose it's up to you if you want to have that type separate.

Reply:
Makes sense. And `declare namespace YouCustomizer` does not work because `Game.YouCustomizer.import` would be a function rather than an attribute, which is not allowed.

Thank you for the feedback! I made the changes. I couldn't think of a better name than just `YouCustomizerT`, though...